lua/lib/tabutil.lua

```diff
@@ -235,5 +235,18 @@ function tab.readonly(params)
   return proxy
 end
 
+--- Create a new table with all values that pass the test implemented by the provided function.
+-- @tparam table t table to check
+-- @param condition callback function that tests all values of provided table
+-- @treturn table table with values that pass the test
+function tab.filter(tbl, condition)
+  local t = {}
+
+  for i=1, #tbl do
```

Comment:
given that tables in lua are a generalized associative structure which is more flexible than the Array type in Javascript it might be more appropriate to use `pairs(...)` to iterate over the table since it will address all values in the table not simply those addressed via positive number or be more explicit in the comment about the expected shape of the table. the `#` operator can produce unexpected results depending on the table:
- http://lua-users.org/wiki/LengthsInLua
- https://www.lua.org/manual/5.1/manual.html#2.5.5

consider the following:
```lua
> t = {1, 2, nil, nil, 5}
> #t
5
> for i,v in ipairs(t) do print(i, v) end
1	1
2	2
> for i,v in pairs(t) do print(i, v) end
1	1
2	2
5	5
> for i=1,#t do print(i, t[i]) end
1	1
2	2
3	nil
4	nil
5	5
```
...its not immediately clear which type of iteration one would association with a function called `filter`. looking at the implementation `select_values` might be a more descriptive name.

Reply:
not sure if @catfact had other thoughts?